deepspeed/runtime/engine.py

```diff
@@ -2908,30 +2916,19 @@ def get_sparse_tensor_module_names(original_set,
                 self.global_steps * self.train_batch_size())
             self.skipped_steps = checkpoint['skipped_steps']
             self.loaded_checkpoint_mp_world_size = checkpoint['mp_world_size']
-            deepspeed_states = [
-                'module',
-                'sparse_tensor_module_names',
-                'skipped_steps',
+            deepspeed_states.extend([
                 'global_steps',
-                'dp_world_size',
+                'skipped_steps',
                 'mp_world_size',
-                'data_sampler',
-                'random_ltd'
-            ]
-        client_state = {}
-
-        if load_lr_scheduler_states:
-            deepspeed_states.append('lr_scheduler')
-        if load_optimizer_states:
-            deepspeed_states.append('optimizer')
+            ])
 
         client_state = {
             key: value
             for key,
             value in checkpoint.items() if not key in deepspeed_states
         }
 
-        if not load_optimizer_states and not load_module_only:
```

Comment:
I believe this is a bug, we should indicate the `optimizer` state was used when `load_optimizer_states=True` and `load_module_only=False`